### ISSUE TITLE
First impressions

### ISSUE THREAD

**Issue by Nicholas-Autio-Mitchell** [idiap/importance-sampling#12]
Hi, thanks for this cool package. I enjoyed reading your paper.

I noticed that Keras >= 2.0.8 is required so that the `clone_model` method is available.

Also, when running your **Quick start** code, the model doesn't learn anything :confused: 
I get an accuracy of 0.09 in all epochs, meaning it is just random selection. I don't have an explanation at the moment. I get the same thing whether I used your ImportanceTraining class or not, so perhaps it isn't related to your code specifically.

**Comment by angeloskath** [comment]
Hi, thanks for the good words.

Good catch! The Keras requirement in the package is >= 2 but I have used `clone_model` in the recent upgrade that added SVRG methods. I should change that in the next release!

That problem is quite odd. I just run the "Quick start" example and the README example and they both run fine. Moreover, each release is being tested on both python 2.7 and 3.4 which you can observe here https://travis-ci.org/idiap/importance-sampling .

It is true that the testing is not as exhaustive as it should be but it should catch obvious bugs or versioning problems. This means that the latest Keras and Tensorflow versions should always work. When using Theano, CNTK or an older version of Keras your mileage may vary.

In case your problem persists, let me know so that we can investigate further. I will leave the issue open for a while, feel free to close it if you fixed it.

Cheers,
Angelos

**Comment by Nicholas-Autio-Mitchell** [comment]
I just ran the code on a different machine, confirmed to be using Python 3.4 and it worked as expected with the following output:

```rust
Epoch 1/5
468/468 [==============================] - 6s 13ms/step - loss: 0.1831 - accuracy: 0.6006 - val_loss: 0.0858 - val_accuracy: 0.9725
Epoch 2/5
468/468 [==============================] - 6s 12ms/step - loss: 0.0395 - accuracy: 0.6228 - val_loss: 0.0675 - val_accuracy: 0.9778
Epoch 3/5
468/468 [==============================] - 6s 12ms/step - loss: 0.0176 - accuracy: 0.6670 - val_loss: 0.0689 - val_accuracy: 0.9812
Epoch 4/5
468/468 [==============================] - 6s 12ms/step - loss: 0.0102 - accuracy: 0.7222 - val_loss: 0.0763 - val_accuracy: 0.9808
Epoch 5/5
468/468 [==============================] - 6s 12ms/step - loss: 0.0057 - accuracy: 0.7747 - val_loss: 0.0934 - val_accuracy: 0.9770
```

I have no explanation for the error I had before, other than I was perhaps using Python 3.5

------------

I found another version issue, which is simple to fix but worth noting in your docs.
So you need Keras 2.0.8 (as discussed above).
I think you also require `Tensorflow >= 1.5` to use the softmax activation as your have in your quick start. This is because the API was changed in 1.5 to specify the `axis` argument. It used to be called `dim`. Keras 2.1.16 also uses `axis`. I'm not sure what combination would make sense...

I guess you could either update version requirements or alter your quick-start code to import Tensoflow (which is basically done anyway with Keras) and call the softmax from there:

```python
import tensorflow as tf

model = Sequential()
...
model.add(Activation(tf.nn.softmax))    # this should always work
```

**Comment by angeloskath** [comment]
I think that the solution is to have the latest **and the minimum** versions being tested in the continuous integration. Specifying a version on TF is beyond the scope of this package since in principle it could even work with Theano or CNTK. So in my opinion it is the Keras version that is incorrect and the tensorflow requirement should be set by Keras.

Regardless, thanks for the good work! I will leave the issue open until I do another release.

**Comment by Nicholas-Autio-Mitchell** [comment]
I understand - it is our of scope for your package.
Please close this issue as soon as you see fit! :smile:

**Comment by angeloskath** [comment]
Still need to change the requirement to Keras >= 2.1 but closing the issue since it doesn't seem to affect people and it also not very focused.
